File: tests/testfiles/fingerprinting/snippets/ai-gen-code/Original_Solution.java
class Solution {
    public long minimumCost(int[] nums) {
        int n = nums.length;
        Arrays.sort(nums);
        int middle = nums[n/2];
        int inc = middle;
        int dec = middle;

        while(!isPal(inc)) inc++; 
        while(!isPal(dec)) dec--;

        return Math.min(calculateCost(nums,inc), calculateCost(nums,dec));
    }


    public boolean isPal(int n) {
        int r, sum = 0, temp;
        temp = n;
        while (n > 0) {
            r = n % 10; 
            sum = (sum * 10) + r;
            n = n / 10;
        }
        return temp == sum;
    }

    public long calculateCost(int[] nums, int r) {
        long cost = 0;
        for (int n: nums) cost += Math.abs(n - r);
        return cost;
    }
}

File: tests/testfiles/fingerprinting/snippets/ai-gen-code/data/find_equalindromic/LeetCode_Solutions/Original_Solution.py
class Solution:
    def minimumCost(self, nums: List[int]) -> int:

        nums.sort()
        n = len(nums)

        def pal(n):
            t = str(n)
            return t == t[::-1]

        def check(n):
            for a in range(n, 0, -1):
                if pal(a):
                    res = sum(abs(it - a) for it in nums)
                    break
            for b in range(n, 10**9):
                if pal(b):
                    res = min(res, sum(abs(it - b) for it in nums))
                    break
            return res

        return min(check(nums[n // 2]), check(nums[n // 2 - 1]))
